python/lsst/daf/butler/core/dimensions/_records.py

```diff
@@ -190,7 +190,7 @@ def direct(
         # This method requires tuples as values of the mapping, but JSON
         # readers will read things in as lists. Be kind and transparently
         # transform to tuples
-        _recItems = {k: v if type(v) != list else tuple(v) for k, v in record.items()}  # type: ignore
+        _recItems = {k: v if not isinstance(v, list) else tuple(v) for k, v in record.items()}  # type: ignore
```

Comment:
I think we deliberately used `type()` here for speed since we know it can only be a list coming from JSON and we need this to be fast. We don't want to worry about subclasses of list here.

Reply:
Latest ruff was complaining about it, should I just suppress this?

Reply:
And why don't we always use `tuple(v)`? `tuple(v)` will return `v` if it's already a tuple.

Reply:
Hmm. That might be a better answer. Let the tuple() constructor deal with it.

Reply:
Ah, not all items are supposed to be tuples there, sorry for the noise.

Reply:
Right, we only want lists to migrate. I think adding a noqa will work but you'll also have to adjust pyproject.toml to stop ruff complaining about unused noqa -- the rubin-env ruff used in Jenkins will not trigger an error and will instead get upset by the unused noqa.